src/core/util.c

```diff
@@ -347,8 +347,11 @@ util_init(void)
 	}
 #endif
 
-#if VG_DRD_ENABLED || VG_HELGRIND_ENABLED
-	_On_drd_or_hg = _On_helgrind + _On_drd;
+#if VG_DRD_ENABLED
+	_On_drd_or_hg += _On_drd;
```

Comment:
_On_drd_or_hg may not be 0 initially.

Instead of this change, maybe just replace:
`_On_drd_or_hg = _On_helgrind + _On_drd;`
with:
`_On_drd_or_hg = On_helgrind + On_drd;`

Reply:
Heh, just to be sure, I checked the C language spec and found that extern variables have static storage duration, so this change is actually correct.

But I think my proposed change is tiny bit cleaner.